devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGradleUtils.java

```diff
@@ -46,44 +46,35 @@ public static String getClassesDir(SourceSet sourceSet, File tmpDir, boolean pop
     }
 
     public static Path mergeClassesDirs(Collection<Path> classesDirs, File tmpDir, boolean populated, boolean test) {
```

Comment:
I cleaned up this method because it was coded with quite a weird pattern that made it hard to understand because we initially had an iterator. Now that we have plain collection, we can use a much simpler approach.